lib/Parse/ParseDecl.cpp

```diff
@@ -5621,8 +5621,47 @@ void Parser::parseAbstractFunctionBody(AbstractFunctionDecl *AFD) {
     Context.Stats->getFrontendCounters().NumFunctionsParsed++;
 
   ParserResult<BraceStmt> Body = parseBraceItemList(diag::invalid_diagnostic);
-  if (!Body.isNull())
-    AFD->setBody(Body.get());
+  if (!Body.isNull()) {
+    // If the body consists of a single expression, turn it into a return 
+    // statement.
+    BraceStmt * BS = Body.get();
+    AFD->setBody(BS);
+    // FIXME: FIXME_NOW: DETERMINE: Do we need to handle the code completion 
+    //        case here?  It is being handled for closure 
+    //        (Parser::parseExprClosure).
+    if (BS->getNumElements() == 1) {
```

Comment:
This code should do the same thing as the closure case: if `Body.hasCodeCompletion()`, it should *not* set the single-expression body bit.  You can copy the comment explaining why here as well.

FYI I'm working on a patch to take return type into account during code-completion of single-expression closures; I can do the same for functions for this PR.

Reply:
Cool!  Okay, the check and comment are there now: 
https://github.com/apple/swift/pull/23251/commits/b5705af35ff6563103d4fcd57183a47d96ddc157 .